frontend/plugincompatibility.lua

```diff
@@ -0,0 +1,458 @@
+--[[--
+Handles plugin version compatibility checks and load overrides.
+
+This module checks if a plugin is compatible with the current KOReader version
+based on the compatibility field in the plugin's _meta.lua file. It also manages
+user overrides for loading incompatible plugins.
+
+Settings structure in G_reader_settings:
+- plugin_load_overrides: {
+    ["plugin-name"] = {
+        action = "always" | "never" | "load-once",
+        version = "plugin_version",
+        koreader_version = "koreader_version",
+    }
+  }
+- plugin_compatibility_prompts_shown: {
+    ["plugin_name-plugin_version-koreader_version"] = true,
+  }
+]]
+
+-- TODO: there must be a way to clean up lingering settings.
+-- e.g. when a plugin is uninstalled, or when a new KOReader version is released
+-- that makes old overrides irrelevant.
```

Comment:
How would you imagine this to be solved @Frenzie ? 

Once per plugin load cycle, check for "garbage" settings entries and delete them? But keep `n+2` where `n` is current version?

Reply:
I'm not entirely sure if I understand what you mean by +2 in this context but in essence sure.